specs/governance/gov-token.md

```diff
@@ -23,50 +75,520 @@
+to another address.
+
+## Definitions
+
+### Voting Power
+
+The amount of governance influence an address has, measured in token units. Voting power is derived from delegated
+token balances and is tracked historically through checkpoints. An address's voting power equals the sum of all tokens
+delegated to it.
+
+### Delegation
+
+The act of assigning one's token balance to another address (or oneself) for voting purposes. Delegation does not
+transfer token ownership or restrict token transfers. Users must explicitly delegate (even to themselves) to activate
+voting power tracking. Delegations are "basic" in that they assign the entire token balance and cannot be
+re-delegated.
+
+### Checkpoint
+
+A data structure recording an address's voting power at a specific block number. Checkpoints are created whenever
+voting power changes due to delegation or token transfers. The structure contains:
+
+- `fromBlock` (uint32): The block number when this voting power became active
+- `votes` (uint224): The voting power amount at that block
+
+## Assumptions
+
+### aGT-001: Owner enforces token inflation schedule
+
+The contract owner (MintManager) is assumed to correctly enforce the predetermined token inflation schedule when
+calling the `mint` function. The GovernanceToken contract itself does not validate minting amounts or timing.
+
+#### Mitigations
+
+- MintManager contract implements inflation schedule logic
+- MintManager ownership is controlled by governance
+- Inflation schedule is publicly documented and auditable
+
+### aGT-002: OpenZeppelin contract implementations are correct
+
+The GovernanceToken relies on OpenZeppelin's implementations of ERC20, ERC20Burnable, ERC20Votes, ERC20Permit, and
+Ownable. These base contracts are assumed to be secure and correctly implemented.
+
+#### Mitigations
+
+- OpenZeppelin contracts are widely audited and battle-tested
+- Specific OpenZeppelin version is pinned in dependencies
+- Contract inheritance is minimal and well-understood
+
+### aGT-003: Users understand delegation requirement for voting
+
+Users are assumed to understand that they must explicitly delegate their tokens (even to themselves) to activate voting
+power tracking. Without delegation, tokens do not contribute to any address's voting power.
+
+#### Mitigations
+
+- Governance interfaces guide users through delegation
+- Documentation clearly explains delegation requirement
+- Delegation status is queryable via `delegates` function
+
+### aGT-004: Governance systems query historical voting power correctly
+
+Governance systems are assumed to correctly use `getPastVotes` with appropriate block numbers to determine voting power
+for proposals. Querying voting power at incorrect blocks could lead to incorrect vote tallying.
+
+#### Mitigations
+
+- Governance contracts specify snapshot blocks in proposals
+- Historical voting power is immutable and deterministic
+- Query functions revert for future blocks
+
+## Dependencies
```

Comment:
devinai Dependencies are only to be used to refer to assumptions or invariants from other files that you depend on for correctness, not generic links to other things